File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 */
public interface BrandService {
    /**
     * 查询所有
     * @return
     */
    public List<TbBrand> findAll1();

    /*
    *品牌分页
    * // pageNum当前页
    *pageSize每页记录数
    * */
    public PageResult findPage(int pageNum,int pageSize);

    /*
    * 添加品牌
    * */
    public void add(TbBrand brand);

    /*
     * 根基 id查询品牌
     * */
    TbBrand findOne(Long id);

    /*
    * 品牌修改
    * */
    public void update(TbBrand brand);

    /*
    * 品牌的删除
    * */
    public void delete(Long []ids);


    /*
     *品牌分页
     * 根据条件查询
     * // pageNum当前页
     *pageSize每页记录数
     * */
    public PageResult findPage(TbBrand brand, int pageNum,int pageSize);

    /*
    * 返回下拉列表
    * */
public  List<Map>selectOptionList();
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//品牌服务
app.service("brandService",function ($http) {
    //查询
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }
    //分页
    this.findPage=function (page, size) {
        return  $http.get("../brand/findPage.do?page="+page+"&size="+size);
    }
    //根据id查询
    this.findOne=function (id) {
        return 	$http.get("../brand/findOne.do?id="+id);
    }
    //添加
    this.add=function (entity) {
        return   $http.post("../brand/add.do",entity);
    }
    //修改
    this.update=function (entity) {
        return   $http.post("../brand/update.do",entity);
    }

    //删除
    this.del=function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    }

    //根据条件分页查询
    this.search=function (page,size,searchEntity) {
        return  $http.post("../brand/search.do?page="+page+"&size="+size,searchEntity)
    }

    //条件下拉列表
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do")
    }

});

